### DISCUSSION TITLE
Share PerspectiveManager between multiple threads in Python?

### DISCUSSION THREAD

**Discussion by Y3NK** [perspective-dev/perspective#1588] [Q&A] [answered]
Hello,

I am wondering if there's a way to communicate through an Perspective Table and maybe the PerspectiveManager between different python programs ?

- on program working as "server", with a PerspectiveManager and a table 
- a second (or more) python program, from which it could read / write to the same table as from the first one (I guess it should pass through the get_table of the PerspectiveManager, but how retrieve the manager of the first program in the second ?)

The 2 programs would run on the same machine.

Have you an idea about how to get that working ?

Many thanks!

**Comment by texodus** [comment, marked as answer]
`perspective-python` comes with both a server (`PerspectiveManager` via tornado or other async server), as well as a [client](https://github.com/finos/perspective/tree/master/python/perspective/perspective/client) which you can use to connect via websocket to a local or remote Python server.

For extreme performance, there are more options for [async](https://perspective.finos.org/docs/md/python.html#async-mode) cooperative GIL-less threading, a [PR for true GIL-less multi-reader support](https://github.com/finos/perspective/pull/1276), and I imagine it is quite easy to get Perspective to work via python's [shared memory support in `multiprocessing`](https://docs.python.org/3/library/multiprocessing.shared_memory.html).  However, none of this is necessary for multiprocessing with perspective - a simple Python client/server will perform quite well without any tweaks.
